Wrappers/Python/test/test_out_in_place.py

```diff
@@ -313,3 +315,132 @@ def test_proximal_out(self):
+        ag_cone_2D.set_angles(angles)
+        ag_cone_2D.set_panel(20)
+
+        ag_cone_3D = AcquisitionGeometry.create_Cone3D(source_position=[0,-10,0],detector_position=[0,10,0])
+        ag_cone_3D.set_angles(angles)
+        ag_cone_3D.set_panel([20,2])
+        
+        self.geometry_test_list = [ag_parallel_2D, ag_parallel_3D, ag_cone_2D, ag_cone_3D]
+        self.data_test_list= [geom.allocate(None) for geom in self.geometry_test_list]
+
+
+    def out_tests(self, processor, data, *args):
+        input=data.copy() #To check that it isn't changed after function calls
+
+        try:
+            processor.set_input(data)
+            out1 = processor.get_output()
+            self.assertDataArraysInContainerAllClose(input, data, rtol=1e-5, msg= "In case processor.set_input(data), processor.get_output() where processor is  " + processor.__class__.__name__+ " the input data has been incorrectly affected by the calculation.")
+        except NotImplementedError:
+            print("get_result test not implemented for  " + processor.__class__.__name__)
+        
+        try:
+            out2 = 0.*(out1.copy())
+            out3 = processor.get_output(out=out2)
+            self.assertDataArraysInContainerAllClose(out1, out2, rtol=1e-5, msg= "Calculation failed using processor.set_input(data), processor.get_output(out=out) where func is  " + processor.__class__.__name__+ ".")
+            self.assertDataArraysInContainerAllClose(input, data,  rtol=1e-5, msg= "In case processor.set_input(data), processor.get_output(out=out) where processor is  " + processor.__class__.__name__+ " the input data has been incorrectly affected by the calculation. ")
+            self.assertDataArraysInContainerAllClose(out1, out3,  rtol=1e-5, msg= "In case processor.set_input(data), output=processor.get_output(out=out) where processor is  " + processor.__class__.__name__+ " the processor supresses the output. ")
+        except NotImplementedError:
+            print("out_test test not implemented for  " + processor.__class__.__name__)
+
+        try:
+            if out1.shape != data.shape:
+                with self.assertRaises(ValueError):
+                    processor.get_output(out=data)
+            else:
+                processor.get_output(out=data)
+                self.assertDataArraysInContainerAllClose(out1, data, rtol=1e-5, msg= "In place calculation failed for processor.set_input(data), processor.get_output(out=data) where processor is  " + processor.__class__.__name__+ "." )
+        except (InPlaceError, NotImplementedError):
+            print("in_place_test test not implemented for  " + processor.__class__.__name__)
+
+    def test_out(self):
+        
+        for geom in self.geometry_test_list:
+            for data_array in self.data_arrays:
+                data=geom.allocate(None)
+                try:
+                    data.fill(data_array)
```

Comment:
I like the testing on the different types of data for each geometry, but maybe add a couple of comments on what this is doing.

Reply:
Added some comments